app/models/reference_payment_settings.rb

```diff
@@ -8,20 +8,23 @@ class ReferencePaymentSettings
                 :deployment_environment
 
   validates :service_id, presence: true
+  validates_with ReferencePaymentValidator
 
   def reference_number_checked?
-    enabled? || ServiceConfiguration.exists?(service_id: service_id, name: 'REFERENCE_NUMBER')
+    reference_number_enabled? || ServiceConfiguration.exists?(service_id: service_id, name: 'REFERENCE_NUMBER')
   end
 
-  def enabled?
+  def reference_number_enabled?
     reference_number == '1'
   end
 
-  def payment_link_checked?
-    payment_enabled? || ServiceConfiguration.exists?(service_id: service_id, name: 'PAYMENT_LINK')
+  def payment_link_url_enabled?
+    payment_link_url_present? || ServiceConfiguration.exists?(service_id: service_id, name: 'PAYMENT_LINK')
   end
 
-  def payment_enabled?
-    false
+  delegate :present?, to: :payment_link_url, prefix: true
```

Comment:
nicely done 😆

Reply:
would this also intercept `present?` called on an instance of `ReferencePaymentSettings`? probably nothing to worry about but if it does and there is an expectation of `@reference_payment_settings.present?` that may return an unexpected result. just a thought

Reply:
Good point we may forget and need to check the presence of reference_payment_settings instance so have put back a full definition